### ISSUE TITLE
False 'collection initializer will implicitly dereference the possibly null member' warning is reproted for collection initializers with extension methods

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#58925]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.1.0 Preview 2.0
VisualStudio.17.Preview/17.1.0-pre.2.0+32104.313
Microsoft .NET Framework
Version 4.8.04084
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
using System.Collections;
using System.Collections.Generic;

_ = new C { Collection = { "", "", "" } };

class C
{
  public MyCollection? Collection { get; set; }
}

internal class MyCollection : IEnumerable
{
  public List<object> List = new();
  public IEnumerator GetEnumerator() => List.GetEnumerator();
}

static class Ext
{
  public static void Add(this MyCollection? collection, object obj) => collection?.List.Add(obj);
}
```

**Expected Behavior**:
No warnings. 
The collection initializer uses the extension method `Ext.Add` which accepts nullable values. The `C.Collection` member is _not_ dereferenced in this code.

**Actual Behavior**:
False `CS8670 ` warning is reported claiming that the collection initializer will implicitly dereference the possibly null member `C.Collection`:
```
Object or collection initializer implicitly dereferences possibly null member 'C.Collection'.
```
The warning is clearly wrong as the member is not dereferenced at all as can be prov,en by running the code. The member is `null` yet the code runs without exceptions, so clearly no dereference happens
